otus-02/src/otus_02/homework/square_code.clj

```diff
@@ -26,6 +28,31 @@
 "vegivenu"
 "sroots  "
 
+(defn remove-spaces [s]
+  (apply str (remove #(= % \space) s)))
+
+(defn normalize-text [text]
+  (-> text
+      (clojure.string/replace #"\s+|\p{Punct}" "")
+      clojure.string/lower-case))
+
+(defn encode-string [input]
+  (let [s (normalize-text input)
+        n (+ 1 (int (Math/floor (Math/sqrt (double (count s))))))
+        rows (partition n n nil s)
+        transposed (map (fn [i] (map #(nth % i \space) rows)) (range n))
+        res3 (clojure.string/join " " (map (fn [elem] (apply str (map str elem))) transposed))]
+    res3))
+
+(defn decode-string [s]
+  (let [a (remove clojure.string/blank? (clojure.string/split s #" "))
+        n (count (first a))
+        transposed (map (fn [i] (map #(nth % i \space) a)) (range n))
+        res (remove-spaces (apply str (map (fn [elem] (apply str (map str elem))) transposed)))]
+    res))
+
```

Comment:
It may work, but.... Looks like you try to use imperative algorithms with mutable state in a case of Clojure, where we prefer immutable data and functional way.
Hope we'll discuss it on Q&A stream